### PR TITLE
Merge Stage to Beta

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by ravinderwalkover — APPROVED** [review]

**Review by shalinee-walkover — APPROVED** [review]
Approve

File: apps/web-giddh/src/app/companyImportExport/component/form/company-import-export-form.html [hunk 1]
```diff
@@ -9,8 +9,10 @@
                     </div>
                     <div class="mr-15" *ngIf="mode === 'export' && fileType === '0'">
                         <label class="d-block">Data Range</label>
-                        <input type="text" name="daterangeInput" daterangepicker [options]="datePickerOptions"
-                        (hideDaterangepicker)="selectedDate($event)" (applyDaterangepicker)="selectedDate($event)" class="form-control" />
+                        <div class="custom-datepicker" (click)="showGiddhDatepicker($event)">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/companyImportExport/component/form/company-import-export-form.html:12]: This story of new datepicker is still under testing, it shouldn't be merged

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/companyImportExport/component/form/company-import-export-form.ts [hunk 2]
```diff
@@ -22,56 +28,61 @@ export class CompanyImportExportFormComponent implements OnInit, OnDestroy {
...
+        private _companyImportExportActions: CompanyImportExportActions,
+        private settingsBranchAction: SettingsBranchActions,
+        private generalService: GeneralService,
+        private modalService: BsModalService,
+        private changeDetectorRef: ChangeDetectorRef) {
+		this.isExportInProcess$ = this._store.pipe(select(s => s.companyImportExport.exportRequestInProcess), takeUntil(this.destroyed$));
+		this.isExportSuccess$ = this._store.pipe(select(s => s.companyImportExport.exportRequestSuccess), takeUntil(this.destroyed$));
+		this.isImportInProcess$ = this._store.pipe(select(s => s.companyImportExport.importRequestInProcess), takeUntil(this.destroyed$));
+		this.isImportSuccess$ = this._store.pipe(select(s => s.companyImportExport.importRequestSuccess), takeUntil(this.destroyed$));
+        this.universalDate$ = this._store.pipe(select(sessionStore => sessionStore.session.applicationDate), distinctUntilChanged(), takeUntil(this.destroyed$));
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/companyImportExport/component/form/company-import-export-form.ts:84]: This is formatted but not visible in GIthub diff view

File: apps/web-giddh/src/app/daybook/advance-search/daybook-advance-search.component.ts [hunk 3]
```diff
@@ -182,24 +196,13 @@ export class DaybookAdvanceSearchModelComponent implements OnInit, OnChanges, On
...
-
-    /**
-     * onDateRangeSelected
-     */
-    public onDateRangeSelected(value) {
-        this.fromDate = moment(value.picker.startDate).format('DD-MM-YYYY');
-        this.toDate = moment(value.picker.endDate).format('DD-MM-YYYY');
-    }
+	public onCancel() {
+		this.fromDate = this.startDate;
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/daybook/advance-search/daybook-advance-search.component.ts:200]: This is formatted but not visible in GIthub diff view

File: apps/web-giddh/src/app/inventory/components/aside-branch-transfer-pane/aside-branch-transfer-pane.component.ts [hunk 4]
```diff
@@ -65,24 +62,12 @@ export class AsideBranchTransferPaneComponent implements OnInit, OnChanges {
...
-			.select(p => p.inventoryInOutState.inventoryUsers.filter(o => o.uniqueName !== this._generalService.companyUniqueName));
-
-		this._store
-			.select(p => p.inventoryInOutState.entryInProcess)
-			.subscribe(p => this.isLoading = p);
-
+		this.stockList$ = this._store.pipe(select(p => p.inventory.stocksList && p.inventory.stocksList.results), takeUntil(this.destroyed$));
+		this.stockUnits$ = this._store.pipe(select(p => p.inventory.stockUnits), takeUntil(this.destroyed$));
+		this.userList$ = this._store.pipe(select(p => p.inventoryInOutState.inventoryUsers.filter(o => o.uniqueName !== this._generalService.companyUniqueName)), takeUntil(this.destroyed$));
+        this._store.pipe(select(p => p.inventoryInOutState.entryInProcess), takeUntil(this.destroyed$)).subscribe(p => this.isLoading = p);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/inventory/components/aside-branch-transfer-pane/aside-branch-transfer-pane.component.ts:69]: This is formatted but not visible in GIthub diff view

File: apps/web-giddh/src/app/search/components/sidebar-components/search.sidebar.component.ts [hunk 5]
```diff
@@ -185,7 +252,28 @@ export class SearchSidebarComponent implements OnInit, OnChanges, OnDestroy {
...
+     * Branch change handler
+     *
+     * @memberof SearchSidebarComponent
+     */
+    public handleBranchChange(selectedEntity: any): void {
+        this.currentBranch.name = selectedEntity.label;
+        this.currentBranchChanged.emit(selectedEntity.value);
+        this.groupService.GetGroupsWithAccounts('', this.currentBranch.uniqueName).subscribe(response => {
+            if (response && response.body && response.body.length) {
+				let accountList = this.flattenGroup(response.body, []);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/search/components/sidebar-components/search.sidebar.component.ts:269]: This is formatted but not visible in GIthub diff view

File: apps/web-giddh/src/app/services/import-excel.service.ts [hunk 6]
```diff
@@ -35,11 +38,14 @@ export class ImportExcelService {
...
 	public processImport(entity: string, model: ImportExcelRequestData) {
 		const companyUniqueName = this._generalService.companyUniqueName;
-		const url = this.config.apiUrl + IMPORT_EXCEL_API.PROCESS_IMPORT
+		let url = this.config.apiUrl + IMPORT_EXCEL_API.PROCESS_IMPORT
 			.replace(':companyUniqueName', companyUniqueName)
 			.replace(':entity', entity)
 			.replace(':isHeaderProvided', model.isHeaderProvided.toString())
-			;
+            ;
+        if (model.branchUniqueName) {
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/services/import-excel.service.ts:46]: Done

File: apps/web-giddh/src/app/settings/other-settings/other-settings.component.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        padding-bottom:5px;
+    }
+}
+
+@media (min-width:1200px) and (max-width:1302px){
+    .other-information {
+        flex: 0 0 66.666667% !important;
+        max-width: 66.666667% !important;
+    }
+    }
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/settings/other-settings/other-settings.component.scss:15]: Done

File: apps/web-giddh/src/app/settings/warehouse/create-warehouse/create-warehouse.component.ts [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+            option.isDefault = false;
+        }
+    }
+    /**
+     * Unsubscribe from all listeners
+     *
+     * @memberof CreateAddressComponent
+     */
+    public ngOnDestroy(): void {
+        document.querySelector('body').classList.remove('fixed');
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/settings/warehouse/create-warehouse/create-warehouse.component.ts:459]: Done

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 9]
```diff
@@ -436,6 +499,22 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
         this.store.pipe(select(state => state.general.openSideMenu), takeUntil(this.destroyed$)).subscribe(response => {
             this.sideBarStateChange(response);
         });
+        this.generalService.isMobileSite.subscribe(s => {
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/header.component.ts:500]: Done

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 10]
```diff
@@ -731,7 +795,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
                 if (isElectron) {
                     this.router.navigate(['/login']);
                 } else {
-                    window.location.href = (environment.production) ? `https://giddh.com/login/?action=logout` : `https://test.giddh.com/login/?action=logout`;
+                    window.location.href = (environment.production) ? `https://stage.giddh.com/login/?action=logout` : `https://test.giddh.com/login/?action=logout`;
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/header.component.ts:796]: Done

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 11]
```diff
@@ -1814,6 +1986,32 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
         }
     }
 
+    /**
+     * Navigates to previous page
+     *
+     * @memberof HeaderComponent
+     */
+    public navigateToPreviousRoute(): void {
+        this.location.back();
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/header.component.ts:1992]: Done

File: apps/web-giddh/src/app/tb-pl-bs/components/export/tb-export-csv.component.ts [hunk 12]
```diff
@@ -92,7 +93,8 @@ export class TbExportCsvComponent implements OnInit, OnDestroy {
 	}
 
 	public ngOnDestroy() {
-		//
+		this.destroyed$.next(true);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/tb-pl-bs/components/export/tb-export-csv.component.ts:96]: This is formatted but not visible in GIthub diff view

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**